lib/src/arithmetic/compound_adder.dart

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// compound_adder.dart
+// Implementation of Compund Integer Adder Module 
+// (Output Sum and Sum1 which is Sum + 1). 
+//
+// 2024 September
+// Author: Anton Sorokin <anton.a.sorokin@intel.com>
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// An abstract class for all compound adder module implementations.
+abstract class CompoundAdder extends Adder {
+
+  /// The addition results (+1) in 2s complement form as [sum1]
+  Logic get sum1 => output('sum1');
+
+  /// Takes in input [a] and input [b] and return the [sum] of the addition
+  /// result and [sum1] sum + 1. 
+  /// The width of input [a] and [b] must be the same.
+  CompoundAdder(super.a, super.b, {super.name}) {
+    if (a.width != b.width) {
+      throw RohdHclException('inputs of a and b should have same width.');
+    }
+    addOutput('sum1', width: a.width + 1);
+  }
+}
+
+/// A trivial compound adder.
+class MockCompoundAdder extends CompoundAdder {
```

Comment:
This isn't really a "mock", since it's functional and synthesizable, just inefficient.  Maybe calling it `TrivialCompoundAdder` (to use your description) or something would be more appropriate?

Reply:
renamed to TrivialCompoundAdder